drivers/serial/uart_nrfx_uarte.c

```diff
@@ -563,6 +567,9 @@ static int uarte_instance_init(struct device *dev,
 #define UARTE_CONFIG_INT(idx)				\
 	.tx_buff_size = UARTE_TX_BUFFER_SIZE(idx)
 
+#define UARTE_IRQ_NUMBER_SET(idx)			\
+	.irq_number = CONFIG_UART_##idx##_IRQ_NUM,
```

Comment:
Instead of adding a dedicated field for this, it would be better to move `irq_enable` to the `UARTE_NRF_IRQ_ENABLED` macro.